heat/core/tiling.py

```diff
@@ -1,8 +1,254 @@
 import torch
 
 from . import dndarray
+from . import factories
 
-__all__ = ["SquareDiagTiles"]
+__all__ = ["SplitTiles", "SquareDiagTiles"]
+
+
+class SplitTiles:
+    def __init__(self, arr):
+        """
+        Initialize tiles with the tile divisions equal to the theoretical split dimensions in
+        every dimension
+
+        Parameters
+        ----------
+        arr : dndarray.DNDarray
+            base array for which to create the tiles
+
+        Examples
+        --------
+        (3 processes)
+        >>> a = ht.zeros((10, 11,), split=None)
+        >>> a.create_split_tiles()
+        >>> print(a.tiles.tile_ends_g)
+        [0] tensor([[ 4,  7, 10],
+        [0]         [ 4,  8, 11]], dtype=torch.int32)
+        [1] tensor([[ 4,  7, 10],
+        [1]         [ 4,  8, 11]], dtype=torch.int32)
+        [2] tensor([[ 4,  7, 10],
+        [2]         [ 4,  8, 11]], dtype=torch.int32)
+        >>> print(a.tiles.tile_locations)
+        [0] tensor([[0, 0, 0],
+        [0]         [0, 0, 0],
+        [0]         [0, 0, 0]], dtype=torch.int32)
+        [1] tensor([[1, 1, 1],
+        [1]         [1, 1, 1],
+        [1]         [1, 1, 1]], dtype=torch.int32)
+        [2] tensor([[2, 2, 2],
+        [2]         [2, 2, 2],
+        [2]         [2, 2, 2]], dtype=torch.int32)
+        >>> a = ht.zeros((10, 11), split=1)
+        >>> a.create_split_tiles()
+        >>> print(a.tiles.tile_ends_g)
+        [0] tensor([[ 4,  7, 10],
+        [0]         [ 4,  8, 11]], dtype=torch.int32)
+        [1] tensor([[ 4,  7, 10],
+        [1]         [ 4,  8, 11]], dtype=torch.int32)
+        [2] tensor([[ 4,  7, 10],
+        [2]         [ 4,  8, 11]], dtype=torch.int32)
+        >>> print(a.tiles.tile_locations)
+        [0] tensor([[0, 1, 2],
+        [0]         [0, 1, 2],
+        [0]         [0, 1, 2]], dtype=torch.int32)
+        [1] tensor([[0, 1, 2],
+        [1]         [0, 1, 2],
+        [1]         [0, 1, 2]], dtype=torch.int32)
+        [2] tensor([[0, 1, 2],
+        [2]         [0, 1, 2],
+        [2]         [0, 1, 2]], dtype=torch.int32)
+        """
+        #  1. get the lshape map
+        #  2. get the split axis numbers for the other axes
+        #  3. build tile map
+        lshape_map = arr.create_lshape_map()
+        tile_dims = torch.zeros((arr.numdims, arr.comm.size), device=arr.device.torch_device)
+        if arr.split is not None:
+            tile_dims[arr.split] = lshape_map[..., arr.split]
+        w_size = arr.comm.size
+        for ax in range(arr.numdims):
+            if arr.split is None or not ax == arr.split:
+                size = arr.gshape[ax]
```

Comment:
to use that function we would need to loop over the number of ranks as well. although that function could be changed to use this method, then it could be used